clang/test/SemaSYCL/intel-fpga-local-ast.cpp

```diff
@@ -0,0 +1,523 @@
+{
+  //CHECK: VarDecl{{.*}}doublepump
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelDoublePumpAttr
+  [[intel::doublepump]] unsigned int doublepump[64];
+
+  //CHECK: VarDecl{{.*}}memory
+  //CHECK: SYCLIntelMemoryAttr
+  [[intel::fpga_memory]] unsigned int memory[64];
+
+  //CHECK: VarDecl{{.*}}memory_mlab
+  //CHECK: SYCLIntelMemoryAttr{{.*}}MLAB
+  [[intel::fpga_memory("MLAB")]] unsigned int memory_mlab[64];
+
+  //CHECK: VarDecl{{.*}}mem_blockram
+  //CHECK: SYCLIntelMemoryAttr{{.*}}BlockRAM
+  [[intel::fpga_memory("BLOCK_RAM")]] unsigned int mem_blockram[32];
+
+  //CHECK: VarDecl{{.*}}reg
+  //CHECK: SYCLIntelRegisterAttr
+  [[intel::fpga_register]] unsigned int reg[64];
+
+  //CHECK: VarDecl{{.*}}singlepump
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelSinglePumpAttr
+  [[intel::singlepump]] unsigned int singlepump[64];
+
+  //CHECK: VarDecl{{.*}}bankwidth
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelBankWidthAttr
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}4
+  //CHECK-NEXT: IntegerLiteral{{.*}}4{{$}}
+  [[intel::bankwidth(4)]] unsigned int bankwidth[32];
+
+  //CHECK: VarDecl{{.*}}numbanks
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelNumBanksAttr
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}8
+  //CHECK-NEXT: IntegerLiteral{{.*}}8{{$}}
+  [[intel::numbanks(8)]] unsigned int numbanks[32];
+
+  //CHECK: VarDecl{{.*}}private_copies
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelPrivateCopiesAttr
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}8
+  //CHECK-NEXT: IntegerLiteral{{.*}}8{{$}}
+  [[intel::private_copies(8)]] unsigned int private_copies[64];
+
+  //CHECK: VarDecl{{.*}}merge_depth
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelMergeAttr{{.*}}"mrg1" "depth"{{$}}
+  [[intel::merge("mrg1", "depth")]] unsigned int merge_depth[64];
+
+  //CHECK: VarDecl{{.*}}merge_width
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelMergeAttr{{.*}}"mrg2" "width"{{$}}
+  [[intel::merge("mrg2", "width")]] unsigned int merge_width[64];
+
+  //CHECK: VarDecl{{.*}}bankbits
+  //CHECK: SYCLIntelNumBanksAttr{{.*}}Implicit{{$}}
+  //CHECK-NEXT: IntegerLiteral{{.*}}16{{$}}
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelBankBitsAttr
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}2
+  //CHECK-NEXT: IntegerLiteral{{.*}}2{{$}}
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}3
+  //CHECK-NEXT: IntegerLiteral{{.*}}3{{$}}
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}4
+  //CHECK-NEXT: IntegerLiteral{{.*}}4{{$}}
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}5
+  //CHECK-NEXT: IntegerLiteral{{.*}}5{{$}}
+  [[intel::bank_bits(2, 3, 4, 5)]] unsigned int bankbits[64];
+
+  //CHECK: VarDecl{{.*}}bank_bits_width
+  //CHECK-NEXT: SYCLIntelNumBanksAttr{{.*}}Implicit{{$}}
+  //CHECK-NEXT: IntegerLiteral{{.*}}4{{$}}
```

Comment:
Likewise - what is this '4'?

Reply:
Same comment is here. We add implict numbank attribute which takes  power of two of `bank_bits' attribute value (2)